test/Benchmarks/src/Startup/Startup.enso

```diff
@@ -6,15 +6,21 @@ polyglot java import java.lang.System as Java_System
 polyglot java import java.io.File as Java_File
 
 type Data
-    Value ~enso_bin:File ~empty_world:File ~hello_world:File
+    Value ~enso_bin:File ~empty_world:File ~hello_world:File ~base_tests:File ~table_tests:File
 
     bench_empty self =
         self.startup [ "--run", self.empty_world.to_text ]
 
     bench_hello self =
         self.startup [ "--run", self.hello_world.to_text ]
 
-    startup self  args =
+    bench_base_tests self =
+        self.startup expected_output="Exiting" [ "--run", self.base_tests.to_text, "exit" ]
```

Comment:
What about adding the `--no-ir-caches` option? Or, a complete alternative:
```
enso --no-ir-caches --no-global-cache --compile test/Table_Tests
```
?

Reply:
I know what current benchmark measures - e.g. _how much time it takes to initialize Table_Tests_ - something people do when they execute `sbt> runEngineDistribution --run test/Table_Tests`.

What would your benchmark measure and why do you think that measurement is more important than the current one?

Reply:
I think having both, with and without caches, would ensure that we don't regress on either?

Reply:
> I know what current benchmark measures - e.g. _how much time it takes to initialize Table_Tests_ - something people do when they execute `sbt> runEngineDistribution --run test/Table_Tests`.
> 
> What would your benchmark measure and why do you think that measurement is more important than the current one?

Isn't the problem with the current benchmark that it could have vastly inconsistent results depending on if IR caches are available/up-to-date?

If I have no caches present, the run will be the same as `--no-ir-caches`. But if there _are_ caches, it will run much faster. Thus we can get different results, suggesting we have performance changes when it may just be environment changes.

If we want to measure the 'cached' performance, we would need to somehow enforce that the environment _does contain correct caches_.

Regardless of the above, like @hubertp IMO having a benchmark of uncached variant (possibly next to the cached one) would still give us valuable insights on the overall performance of the compiler (which I think could be very useful in the future e.g. when I'll be integrating the typechecker and will want to see how much it influences the compilation time). In the cached variant, changes like the typechecker are likely to be not visible, if their IR gets correctly cached (so we'd only measure the cache loading time - also a very useful metric, but completely different one).

Reply:
Btw. this is for a separate thing, but could we gather statistics of how much time we spend in _each_ of the compiler passes that we have during the compilation? It feels like it could help us find any bottlenecks.

I know we can do this using a profiler, but I'm wondering if we could gather this kind of info as part of the nightly benchmark job to also be able to track how (if?) it changes over time.

Reply:
> track how (if?) it changes over time.

That's a subject of #8419

> could we gather statistics of how much time we spend in _each_ of the compiler passes

one we start to benchmark the `runtime-compiler` we'll certainly follow up with fixes and deep dive into individual passes.

Reply:
> Isn't the problem with the current benchmark that it could have vastly inconsistent results depending on if IR caches are available/up-to-date?

I think caches are going to be created during the first warmup run.

> If I have no caches present, the run will be the same as `--no-ir-caches`. 

No, that's not. Users always have caches inside of standard libraries. That's why I am hesitating to use `--no-ir-caches` - it doesn't match a typical user use-case.

> insights on the overall performance of the compiler

That's the goal of #8419, not this PR.

Reply:
> having both, with and without caches, would ensure that we don't regress on either?
> --no-ir-caches

Created [wip/jtulach/Benchmark_Real_Startup_No_ir_Caches](https://github.com/enso-org/enso/compare/wip/jtulach/Benchmark_Real_Startup...wip/jtulach/Benchmark_Real_Startup_No_ir_Caches). The [option is added](https://github.com/enso-org/enso/commit/7a5097fcd97a1281abf56f358bf7fe039e73d9dc) and the [benchmark is running](https://github.com/enso-org/enso/actions/runs/7621136673)...

> hesitating to use --no-ir-caches - it doesn't match a typical user usage.

...however I am not sure what to expect. It will just _"show some number"_.

Reply:
Two more benchmark runs scheduled:
- run [with normal caching](https://github.com/enso-org/enso/actions/runs/7683825375)
- run [without any caching](https://github.com/enso-org/enso/actions/runs/7683828597)

As far as I can tell the differences are significant `4.5s` vs `16s`. That nicely shows the _benefit of caching_. Not sure how relevant that is for `runtime/compiler` benchmarking, however.